karapace/backup/api.py

```diff
@@ -390,14 +378,24 @@ def _handle_restore_topic(
         raise BackupTopicAlreadyExists(f"Topic to restore '{instruction.topic_name}' already exists")
 
 
+producer_exception: Exception | None = None
+
+
 def _raise_backup_error(exception: Exception) -> NoReturn:
-    raise BackupDataRestorationError("Error while producing restored messages") from exception
+    global producer_exception  # pylint: disable=global-statement
+    producer_exception = exception
+
+
+def check_producer_exception() -> None:
+    if producer_exception is not None:
+        raise BackupDataRestorationError("Error while producing restored messages") from producer_exception
 
 
 def _handle_producer_send(
     instruction: ProducerSend,
     producer: KafkaProducer,
 ) -> None:
+    check_producer_exception()
```

Comment:
issue: Let's document (with comment) the reason why we're also doing this here. Is it to reach a stop quicker than waiting for the call in `restore_backup()`?